scripts/us_eia/opendata/process/common.py

```diff
@@ -0,0 +1,167 @@
+"""Process EIA datasets to produce TMCF and CSV."""
+
+from sys import path
+# For import util.alpha2_to_dcid
+path.insert(1, '../../../../')
+
+import csv
+import json
+import logging
+import pandas as pd
+import util.alpha2_to_dcid as alpha2_to_dcid
+from collections import defaultdict
+
+
+_COLUMNS = ['place', 'stat_var', 'date', 'value', 'unit', 'scaling_factor',
+            'eia_series_id']
+
+_TMCF_STRING = """
+Node: E:EIATable->E0
+typeOf: dcs:StatVarObservation
+observationAbout: C:EIATable->place
+variableMeasured: C:EIATable->stat_var
+observationDate: C:EIATable->date
+value: C:EIATable->value
+unit: C:EIATable->unit
+scalingFactor: C:EIATable->scaling_factor
+eiaSeriesId: C:EIATable->eia_series_id
+"""
+
+_QUARTER_MAP = {
+    'Q1': '03',
+    'Q2': '06',
+    'Q3': '09',
+    'Q4': '12',
+}
+
+
+def _parse_date(d):
+  if len(d) == 4:
+    # Yearly
+    return d
+
+  if len(d) == 6:
+    yr = d[:4]
+    m_or_q = d[4:]
+
+    if m_or_q.startswith('Q'):
+      # Quarterly
+      if m_or_q in _QUARTER_MAP:
+        return yr + '-' + _QUARTER_MAP[m_or_q]
+    else:
+      # Monthly
+      return yr + '-' + m_or_q
+
+  return None
+
+
+def _eia_dcid(raw_sv):
+  return 'dcid:eia/' + raw_sv.lower()
+
+
+def _print_stats(stats):
+  print('\nSTATS:')
+  for k, v in stats.items():
+    print(f"\t{k} = {v}")
+  print('')
+
+
+def _find_dc_place(raw_place, stats):
+  # At the moment, we only support states and US.
+  if raw_place == 'US':
+    return 'country/usa'
+  elif raw_place in alpha2_to_dcid.USSTATE_MAP:
+    return alpha2_to_dcid.USSTATE_MAP[raw_place]
+
+  stats['error_unsupported_places'] += 1
+  return None
+
+
+def _generate_default_statvar(raw_sv, sv_map):
+  if raw_sv in sv_map:
+    return
+  raw_sv_id = _eia_dcid(raw_sv)
+  sv_map[raw_sv] = '\n'.join([
+      f"Node: {raw_sv_id}",
+      'typeOf: dcs:StatisticalVariable',
+      'populationType: schema:Thing',
+      f"measuredProperty: {raw_sv_id}",
+      'statType: dcs:measuredValue',
+  ])
+
+
+def process(in_json, out_csv, out_sv_mcf, out_tmcf, extract_place_statvar_fn,
+            generate_statvar_schema_fn):
+  """
```

Comment:
maybe describe the requirements for the lambda functions (even with [typing](https://stackoverflow.com/questions/33833881/is-it-possible-to-type-hint-a-lambda-function) it might not be too clear)

Reply:
Added more details of args / return value for the lambda functions